### DISCUSSION TITLE
Custom sorting for global search categories

### DISCUSSION THREAD

**Discussion by fblaser** [filamentphp/filament#12954] [Ideas]
Currently, the global search results in a filament panel are sorted alphabetically by model name and there is no way to influence this. 

**The problem:** 
In one of our projects, we have Invoices and Services.  Both models are globally searchable, but a matching "Service" is more interesting that a matching Invoices, so I would like to display the Services first. 

**Possible solutions:**
1) Add a new optional attribute in Resource named `$globalSearchCategoryLabel` or something similar and then sorting the categories alphabetically, giving the developper control over the sort order of the categories by changing the label. 
2) Add a new optional attribute in Resource named `$globalSearchSort`, which would work similarly to $navigationSort for navigation links.

I would gladly work on a PR if that sounds like something that would benefit the project.

**Comment by bernhardh** [comment]
Is there already a solution for this?

Related other discussion https://github.com/filamentphp/filament/discussions/5618 I found

**Reply by fblaser** [reply]
To my knowledge, this is still open. 

Currently, PRs for version 3 are not accepted anymore. If this is not included in v4, I'll work on it when it's released.

**Reply by fblaser** [reply]
@bernhardh: Which of the two proposed ideas would help in your case?

**Reply by bernhardh** [reply]
Option 2 makes much more sense, in my opinion. With Option 1, you'd be forced to choose the labels based on sorting rather than what fits best.

My current scenario: I have Countries, Cities, Articles (in exactly this sorting priority). There are way less countries, than cities, than articles. With your option 1 it would sort after Articles, Cities and Countries.

**Reply by fblaser** [reply]
Agreed, option 2 would work best in my case as well.

**Reply by fblaser** [reply]
If I'm not mistaken, PR are not possible on v3 anymore. I'll wait and see what's in v4 to work on this.

**Comment by raobilal482** [comment]
both of the options are not working for sorting global search

**Reply by fblaser** [reply]
Yes, you're browsing the "ideas" discussion, this is not implemented yet :)

**Comment by joe-herbert** [comment]
I've just come across this problem, is there a workaround until the above is implemented or is it simply impossible for now?

**Reply by fblaser** [reply]
I think it's simply impossible for now. 

@danharrin : Would it be ok if I submit a PR for this for v3 even though PRs for v3 have been discouraged?

**Reply by danharrin** [reply]
Check out the `DefaultGlobalSearchProvider` class, copy it, and pass the class name to `$panel->globalSearch()` in the config. Then you can override the order of resources.

I will accept a PR for providing a `$globalSearchSort` for resources in global search in v4, but this can be done in your app.

**Reply by fblaser** [reply]
Awesome, thanks for the workaround!

**Comment by fblaser** [comment]
This has been solved by #18159
